frontend/src/components/teacher/session-creation/DistributeAssessmentButton.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { useMutation } from "@apollo/client";
+import { Button } from "@chakra-ui/react";
+
+import { CREATE_TEST_SESSION } from "../../../APIClients/mutations/TestSessionMutations";
+import { GET_TEST_SESSIONS_BY_TEACHER_ID } from "../../../APIClients/queries/TestSessionQueries";
+import type { TestSessionRequest } from "../../../APIClients/types/TestSessionClientTypes";
+import { PaperPlaneOutlineIcon } from "../../../assets/icons";
+import { DISPLAY_ASSESSMENTS_PAGE } from "../../../constants/Routes";
+import { generateAccessCode } from "../../../utils/TestSessionUtils";
+import useToast from "../../common/info/useToast";
+
+import DistributeAssessmentModal from "./DistributeAssessmentModal";
+
+interface DistributeAssessmentButtonProps {
+  testSession: Omit<TestSessionRequest, "accessCode">;
+}
+
+const DistributeAssessmentButton = ({
+  testSession,
+}: DistributeAssessmentButtonProps): React.ReactElement => {
+  const [showDistributeModal, setShowDistributeModal] = React.useState(false);
+
+  const history = useHistory();
+  const { showToast } = useToast();
+
+  const [createSession, { loading }] = useMutation<{
+    createSession: string;
+  }>(CREATE_TEST_SESSION, {
+    refetchQueries: [{ query: GET_TEST_SESSIONS_BY_TEACHER_ID }],
+  });
+
+  const handleCreateSession = async () => {
+    try {
+      await createSession({
+        variables: {
+          testSession: { ...testSession, accessCode: generateAccessCode() },
```

Comment:
my instinct would be to put this on the backend actually but I can't think of a concrete reason why. I suppose a bad actor could probably break our infra by spamming the same code over and over but I think this is ok